hack/release-tools/tag-release.sh

```diff
@@ -99,8 +113,10 @@ if [[ "$VELERO_PATCH" > 0 ]]; then
 
     # TODO can/should we add a way to review the cherry-picked commits before the push?
```

Comment:
Given that the default behaviour for this script is to prepare the branch without pushing, perhaps this comment could be removed as there is now an opportunity to review the cherry-picked commits.

Reply:
Good point. I think there's still more to be done for the patch release process, as my original intention was to somehow provide a list of commit hashes or PR numbers and have the script cherry-pick them for the user. This may not be reasonable given potential conflicts, but it's also something I don't really want to fix in this PR.

Reply:
I agree, that seems out of scope for this PR. In looking at this script earlier, I think there may be some issues with the patch release section to do with checking for the presence of the release branch, but also when a release branch is tracked in more than one remote (as would be case if you have remotes for your own fork and upstream). I'll file an issue for those to address for 1.5.1.

Reply:
Yeah, I'm sure there are some weird cases. I'm definitely tracking release branches on my own fork and on the upstream remote, but when testing this I added a 3rd remote which was a duplicate of my fork so I didn't make any bad releases against real upstream.